android/src/main/java/com/onegini/mobile/sdk/reactnative/handlers/pins/PinAuthenticationEventEmitter.kt

```diff
@@ -7,48 +7,32 @@ import com.onegini.mobile.sdk.reactnative.Constants.PinFlow
 import com.onegini.mobile.sdk.reactnative.OneginiComponents.reactApplicationContext
 import com.onegini.mobile.sdk.reactnative.exception.OneginiWrapperErrors
 
-
 class PinAuthenticationEventEmitter() {
   fun onPinOpen(profileId: String) {
     val dataMap = Arguments.createMap()
     dataMap.putString("action", Constants.PIN_NOTIFICATION_OPEN_VIEW)
     dataMap.putString("flow", PinFlow.Authentication.toString())
     dataMap.putString("profileId", profileId);
     reactApplicationContext.getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter::class.java)
-      .emit(Constants.ONEWELCOME_PIN_NOTIFICATION, dataMap)
+      .emit(Constants.ONEWELCOME_PIN_AUTHENTICATION_NOTIFICATION, dataMap)
   }
 
   fun onPinClose() {
     val dataMap = Arguments.createMap()
     dataMap.putString("action", Constants.PIN_NOTIFICATION_CLOSE_VIEW)
     dataMap.putString("flow", PinFlow.Authentication.toString())
     reactApplicationContext.getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter::class.java)
-      .emit(Constants.ONEWELCOME_PIN_NOTIFICATION, dataMap)
+      .emit(Constants.ONEWELCOME_PIN_AUTHENTICATION_NOTIFICATION, dataMap)
   }
 
-  fun onError(errorCode: Int, errorMessage: String) {
-    val data = Arguments.createMap()
-    data.putString("action", Constants.PIN_NOTIFICATION_SHOW_ERROR)
-    data.putString("flow", PinFlow.Authentication.toString())
-    data.putInt("errorType", errorCode)
-    data.putString("errorMsg", errorMessage)
-    reactApplicationContext.getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter::class.java)
-      .emit(Constants.ONEWELCOME_PIN_NOTIFICATION, data)
-  }
-  // TODO: This is a remnant from some hacky old code, let's refactor this :^) RNP-98
-  // This isn't the most logical way to send the remaining attempts to the plugin,
-  // but I did it to not have to modify iOS/JS parts as well
-  fun onWrongPin(remainingAttempts: Int) {
-    val userInfo = Arguments.createMap()
-    userInfo.putString("remainingFailureCount", remainingAttempts.toString())
-
+  fun onIncorrectPin(remainingAttempts: Int) {
     val dataMap = Arguments.createMap()
-    dataMap.putString("action", Constants.PIN_NOTIFICATION_SHOW_ERROR)
```

Comment:
In order not to minimize boilerplate you can use Koltin `.apply` and `also` scope function https://kotlinlang.org/docs/scope-functions.html#apply
So you're gonna have
```
Arguments.createMap()
.apply {
putString("action", Constants.PIN_NOTIFICATION_SHOW_ERROR)
putString("flow", Constants.PinFlow.Authentication.toString())
putMap("userInfo", userInfo)
putString("action", Constants.PIN_NOTIFICATION_INCORRECT_PIN)
putString("flow", PinFlow.Authentication.toString())
putString("remainingFailureCount", remainingAttempts.toString())
putInt("errorType", OneginiWrapperErrors.WRONG_PIN_ERROR.code.toInt())
putString("errorMsg", OneginiWrapperErrors.WRONG_PIN_ERROR.message)
}
.also { dataMap ->
reactApplicationContext.getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter::class.java)
      .emit(Constants.ONEWELCOME_PIN_NOTIFICATION, dataMap)
      .emit(Constants.ONEWELCOME_PIN_AUTHENTICATION_NOTIFICATION, dataMap)
}
```